flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -113,39 +108,6 @@
  */
 public class SourceStreamTaskTest extends SourceStreamTaskTestBase {
 
-    @Test
-    public void testInputEndedBeforeStopWithSavepointConfirmed() throws Exception {
```

Comment:
test removal explanation: This situation can no longer happen, as we first stop the source when we see a trigger for synchronous checkpoint.